pkg/tsdb/cloudwatch/metric_find_query.go

```diff
@@ -172,7 +172,7 @@ var dimensionsMap = map[string][]string{
 	"AWS/EC2CapacityReservations": {"CapacityReservationId"},
 	"AWS/EC2Spot":                 {"AvailabilityZone", "FleetRequestId", "InstanceType"},
 	"AWS/ECS":                     {"ClusterName", "ServiceName"},
-	"AWS/EFS":                     {"FileSystemId"},
+	"AWS/EFS":                     {"FileSystemId", "StorageClass"},
```

Comment:
Why is `StorageClass` required as a dimension? From https://docs.aws.amazon.com/efs/latest/ug/efs-metrics.html:

> Amazon EFS metrics use the EFS namespace and provide metrics for a single dimension, FileSystemId.

Reply:
Had some time to verify this today. Indeed, `StorageClass` is a possible dimension when using the `StorageBytes` metric.

![Screenshot from 2021-05-18 13-33-44](https://user-images.githubusercontent.com/4025665/118662000-f0cd8680-b7ef-11eb-9998-f837acc73f4e.png)

The problem is that this dimension is not available for any of the other metrics so displaying it for the whole namespace can be confusing. IMO I would not add it because people can still manually write the dimension in this case (and won't hurt the rest of metrics). WDYT @sunker?

Ideally, I would create a separate map for metrics dimensions (I don't know if there are more cases like this) but that would require changes in the frontend as well since the metric is not part of the query when resolving these dimensions. I'd be happy to do so in a separate PR if you think so.

Reply:
I agree that it's kinda confusing having a dimension that only applies to specific metrics and that the UI would need to change to make that clear.

I also didn't realise that dimensions and metrics could be typed directly in

Reply:
I checked with @sunker and we are okay merging this as it is and create a follow-up feature request to be able to have dimensions per metric. Thanks again @freshleafmedia!